internal/pointer/bool.go

```diff
@@ -0,0 +1,13 @@
+package pointer
```

Comment:
please add tests for this package and comment of what the functions are doing. Examples how to use them. etc.